tests/baselines/reference/conditionalTypes1.types

```diff
@@ -1010,8 +1010,8 @@ type Bar<T> = T extends string ? boolean : number;
 >T : T
 
 const convert = <U>(value: Foo<U>): Bar<U> => value;
->convert : <U>(value: Foo<U>) => Foo<U>
-><U>(value: Foo<U>): Bar<U> => value : <U>(value: Foo<U>) => Foo<U>
+>convert : <U>(value: Foo<U>) => Bar<U>

```

Comment:
So the side-effect of this change is that identical conditionals aren't cached anymore, since we don't know they're identical. Hm. Ah well. Same's true of signatures.